### PR TITLE
Bug 1173104 - [Contacts][NGA] Extract 'isEmpty' out of contacts.js

### PR DESCRIPTION
**Pull request by fcampo**

### CODE REVIEW

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=f90ef7706babc85367c4f691259dc7fa43d14840

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=c9c51f92ff5ad16e1fe631afe2174d472903044b

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=7f2f9b2dce2132a2008c27f71a90ce7546ab818d

File: apps/communications/contacts/js/bootstrap.js [hunk 1]
```diff
@@ -33,6 +33,7 @@
        '/shared/js/contacts/import/utilities/config.js',
        '/contacts/js/utilities/extract_params.js',
        '/contacts/js/utilities/cookie.js',
+       '/contacts/js/utilities/object.js',
```
Comment (arcturus) [inline apps/communications/contacts/js/bootstrap.js]: We don't need to load this in the critical path, the function `isEmpty` is being used on the views:
detail
form
settings

So shouldn't need to be loaded on the bootstrap to save startup time.

File: apps/communications/contacts/js/utilities/object.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
+'use strict';
+
+(function(exports) {
+
+  exports = exports || {};
+
+  var object = {
+    // Checks if an object fields are empty, by empty means
+    // field is null and if it's an array it's length is 0
+    isEmpty: function (obj, fields) {
```
Comment (arcturus) [inline apps/communications/contacts/js/utilities/object.js]: I'm wondering if we should move this to the`ContactsService` that already has a method isEmpty to know if we don't have contacts. But definitely is somehow associated to know something from a mozcontact

**Comment by arcturus** [comment]
ni: The naming `object` is too generic, either we can append to the `ContactsService` or use a different naming.

Perhaps `utils/mozContacts` ?
@borjasalguero  wdyt?

File: apps/communications/contacts/js/utilities/object.js (hunk 2 above)
Comment (fcampo) [inline apps/communications/contacts/js/utilities/object.js]: `ContactService.isEmpty` checks if the DataBase is empty. This functions checks if an object (any) have empty fields, so I don't think putting it in ContactService is a good idea.

Plus, `ContactService` will be Thread.js in the near future, so no reason to add anything there.

File: apps/communications/contacts/js/bootstrap.js (hunk 1 above)
Comment (fcampo) [inline apps/communications/contacts/js/bootstrap.js]: Agree. I didn't want to make more than minimal changes for this, but I also think it's better to load it when needed.

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=d9e9a9f5b8f6a7316a10caacde8293a210cd36c5

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=22486062c9e57d47464b55a36cd676af9e69660c

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=813c7aab8cc45902387bf371d7fd385b20dd1faa

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=18caf3010cfe41c3bae36f49616f61d74339b9d0

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=6a365529f8ff9e70e7fe1dc3b743d4108d01e966

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=3c5868b1717ad2189f4ef6a6ec9619b24bd940f7

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=d5ad8f1a431187b1f88ab442786ecc58518b6467

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=60159439147027cdef28afdb01e7534e1f60863b

**Pull request merged**